### PR TITLE
GEOPY-1335: improve drillholes tables memory and access

### PR DESCRIPTION
**Pull request by MatthieuCMira**
**GEOPY-1334 - improve drillholes tables memory and access*
update drillholes table based on Ben's comment:

- do not have the table on memory
- access the name directly, without loading everything
- update the concatenator data and index.

### CODE REVIEW

**Comment by domfournier** [comment]
I think that's supposed to be GEOPY-1335?

File: geoh5py/shared/concatenation/drillholes_group_table.py [hunk 1]
```diff
@@ -70,20 +71,68 @@ def _create_structured_array(output: np.ndarray, names: tuple[str]) -> np.ndarra
...
+            # get the names of the data
+            if drillholes:
+                names = ("Drillhole",) + names
+
+            # transform to a structured array
+            return self._create_structured_array(
+                np.concatenate(all_data_list, axis=0), names
+            )
+
+        raise ValueError("No drillhole found in the concatenator.")
```
Comment (domfournier) [inline geoh5py/shared/concatenation/drillholes_group_table.py:120]:
```suggestion
        if self.index_by_drillhole is None:
            raise ValueError("No drillhole found in the concatenator.")
            
        all_data_list = []
        for object_, data_dict in self.index_by_drillhole.items():
            data_list: list = []
            no_data_values: list = []
            for name, info in data_dict.items():
                if name in names:
                    data_list.append(
                        self.parent.data[name][info[0] : info[0] + info[1]]
                    )
                    no_data_values.append(self.nan_value_from_name(name))

            data_list = self._pad_arrays_to_first(data_list, no_data_values[1:])

            if drillholes:
                # add the object list to the first position of the data list
                data_list.insert(0, [object_] * data_list[0].shape[0])

            # create a numpy array
            all_data_list.append(np.array(data_list, dtype=object).T)

        # get the names of the data
        if drillholes:
            names = ("Drillhole",) + names

        # transform to a structured array
        return self._create_structured_array(
            np.concatenate(all_data_list, axis=0), names
        )

```

**Review by domfournier — COMMENTED** [review]
Small comment to simpify indentation. Looks great.

**Comment by MatthieuCMira** [comment]
Nothing to add, can be merged?

The type error is not possible, but is needed for pylint

**Review by domfournier — APPROVED** [review]
Looks good.

**Pull request merged**